### ISSUE TITLE
[Feature] Enhanced Course Preview and Path Planning

### ISSUE THREAD

**Issue by geiras** [Courseplay/CourseGenerator#27]
**What is the problem this feature/enhancement solves?**

Currently, it is difficult to predict the course that a worker will take in Course Play before it begins. This is due to the unclear path that is displayed on the map, and the number of intersecting lines that can appear. As a result, it can be challenging to understand how changes in settings will impact the path that the worker will take.

**Describe the feature that you'd like**

To address this issue, I propose the addition of a feature that allows players to better preview the course before it is generated. This could be done in the form of an animated example of the course from start to finish with all the lines it will follow. Alternatively, if an animated preview is not feasible due to processing time constraints, a better static preview could be displayed.

This would allow players to see where the course starts, how the path line animates between each point, and ultimately get a better idea of what route the worker will take. Additionally, an option to display numbers at each point would be helpful too. This would make it easier to identify specific points on the course and understand how they relate to the overall path.

**Describe alternatives you've considered**

As an alternative to animating the course path, a simple preview could be displayed showing the start and end points, as well as intermediate turning points. This would provide a basic overview of the course without the need for animation.

Another alternative would be to add a toggle switch that allows players to hide or show numbers instead of arrows on each intermediate point.

**Additional context**

I believe that providing a better way to preview courses would be a valuable addition to Course Play. This would make it easier for players to create efficient and effective courses, and would also help to reduce frustration when trying to understand how settings affect the path that the worker will take.

**Comment by Tensuko** [comment]
Animation is almost impossible.
We have the Plan for the next generator to add arrows to the path but more cant be done.
A preview wont be done as well. Makes no sence to generate a preview before generate the course. But I want a bigger help menu entry that explain the patterns more with examples.

**Comment by geiras** [comment]
Indeed, the base game appears somewhat limited in terms of what one can achieve with the mods windows, and I was aware that implementing animation would be rather complex, even if it would ideally assist players in comprehending the course.

It is reassuring to know that you also plan for the upcoming generator to incorporate arrows within the course itself, although I understand that achieving more than this, such as a preview function, may prove difficult.

However, allow me to politely disagree when you claim that generating a preview before creating the course is nonsensical. I believe it does make sense because initially, I would adjust an option, then generate the course and then attempt to discern what had changed. I would then modify another option and generate the course once more. Essentially, this would eliminate my need to continuously generate courses simply to preview the path. If you remove this necessity for users and provide a preview sufficient for them to understand the path, it seems quite logical from a user experience perspective, which happens to be my area of expertise.

Regarding a better explanation about patterns and such, as well as an expanded help menu, I do think these additions will prove helpful. However, they may not entirely replace a quality course preview since factors such as field shape and other variables will differ. Nevertheless, it is undoubtedly a step forward.

Thank you for your efforts in making the AI slightly more usable in-game. I truly appreciate your hard work and my suggestions are of course aimed at improving the game, while understanding the challenges you may face. It is ultimately your decision how things proceed – my comments merely serve to share some frustrations experienced with the course, which are by no means comparable to those encountered with the original in-game AI. Even if nothing changes, I shall remain quite content as a player.

**Comment by pvaiko** [comment]
> If you remove this necessity for users and provide a preview sufficient for them to understand the path, it seems quite logical from a user experience perspective, which happens to be my area of expertise.

We may be thinking of different things when we say preview. What exactly you mean by a preview? Currently, in order to show any kind of preview, we have to generate the course completely anyway. 

Also, if you had a few, examples of exactly what information you expect to get from a preview, that would help. Like a screenshot of the current one, with a short explanation of what is not clear there.

**Comment by rohne83** [comment]
Only thing I could imagine for preview (and in this case quite helpful indeed) would be a simple preview of the fieldrow angles when the angle is set manually by user. Especially on large fields with complicated shapes it can be very frustrating when you calculate the course with a set angle, have to wait several minutes for the course to be generated, and then you see that you should have added/substracted 5 or 10 degrees more to get a good course, and have to recalculate with long waiting time again.
